fe/fe-core/src/main/java/org/apache/doris/analysis/DateLiteral.java

```diff
@@ -361,8 +361,18 @@ public int compareLiteral(LiteralExpr expr) {
         if (expr == MaxLiteral.MAX_VALUE) {
             return -1;
         }
+
+        long thisExprLongValue = getLongValue();
+        long otherExprLongValue = expr.getLongValue();
+        if (!getType().equals(expr.getType())) {
+            if (getType().equals(Type.DATE)) {
+                thisExprLongValue *= 1000000;
```

Comment:
Better define this `1000000` as a final static variable, and explain what this number's mean.

Reply:
Thanks for your reply.
I have modified it according to your suggestion.